### PR TITLE
Update opendistro-for-elasticsearch-release-notes-1.10.1.md for more syntax fixes

### PR DESCRIPTION
**Pull request by peterzhuamazon**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/opendistro-infra/issues/255

*Description of changes:*
This PR is to update opendistro-for-elasticsearch-release-notes-1.10.1.md for more syntax fixes

*Test Results:*
No Need

**Note: If this PR is related to Helm, please also update the README for related documentation changes. Thanks.**
**https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/helm/README.md**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: release-notes/opendistro-for-elasticsearch-release-notes-1.10.1.md [hunk 1]
```diff
@@ -368,9 +368,9 @@ Open Distro for Elasticsearch 1.10.1 includes the following features, enhancemen
 
 ### SQL
 
-* update user documentation for testing odbc driver connection on windows([#722](https://github.com/opendistro-for-elasticsearch/sql/pull/722))
-* Added workaround for identifiers with special characters in troubleshooting page([#718](https://github.com/opendistro-for-elasticsearch/sql/pull/718))
-* Update release notes for OD 1.10 release([#699](https://github.com/opendistro-for-elasticsearch/sql/pull/699))
+* update user documentation for testing odbc driver connection on windows ([#722](https://github.com/opendistro-for-elasticsearch/sql/pull/722))
```
Comment (gaiksaya) [inline release-notes/opendistro-for-elasticsearch-release-notes-1.10.1.md:371]: Do we need to capitalize the first letter of all sentences or take them as submitted?

Comment (peterzhuamazon) [reply release-notes/opendistro-for-elasticsearch-release-notes-1.10.1.md:371]: I just take them as submitted, it is the same procedure as before, no problems

Comment (camerski) [reply release-notes/opendistro-for-elasticsearch-release-notes-1.10.1.md:371]: Please fix the grammar. "Update", "ODBC", and "Windows" should be capitalized.

File: release-notes/opendistro-for-elasticsearch-release-notes-1.10.1.md [hunk 2]
```diff
@@ -368,9 +368,9 @@ Open Distro for Elasticsearch 1.10.1 includes the following features, enhancemen
...
+* Added workaround for identifiers with special characters in troubleshooting page ([#718](https://github.com/opendistro-for-elasticsearch/sql/pull/718))
```
Comment (camerski) [inline release-notes/opendistro-for-elasticsearch-release-notes-1.10.1.md:372]: Change "Added" to "Add" so the tense is the same as the other sentences.

**Review by camerski — CHANGES_REQUESTED** [review]

**Review by camerski — APPROVED** [review]
I will do another pass today to correct grammar

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**